internal/transport/http_util.go

```diff
@@ -332,19 +334,33 @@ func (w *bufWriter) Write(b []byte) (n int, err error) {
 		n, err = w.conn.Write(b)
 		return n, toIOError(err)
 	}
+	if w.buf == nil {
+		b := w.pool.Get().(*[]byte)
+		w.buf = *b
+	}
 	for len(b) > 0 {
-		nn := copy(w.buf[w.offset:], b)
+		nn := copy((w.buf)[w.offset:], b)
 		b = b[nn:]
 		w.offset += nn
 		n += nn
 		if w.offset >= w.batchSize {
-			err = w.Flush()
+			err = w.flush()
 		}
 	}
 	return n, err
 }
 
 func (w *bufWriter) Flush() error {
+	err := w.flush()
+	if w.buf != nil {
+		b := w.buf
+		w.pool.Put(&b)
```

Comment:
`b := w.buf` line is important as otherwise we keep modifying the same buf after putting it to the pool